torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2897,7 +2901,8 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
                     # reduce_dtype matches the param dtype.
                     param.grad.data = param.grad.data.to(self.mixed_precision.reduce_dtype)
 
-                if self.gradient_predivide_factor > 1:
+                if self.gradient_predivide_factor > 1 and \
```

Comment:
I see, so default comm hook is none when sharded, and the default hook when unsharded, so the old behavior is exactly retained